app/models/agents/weibo_publish_agent.rb

```diff
@@ -76,6 +88,24 @@ def publish_tweet text
       weibo_client.statuses.update text
     end
 
+    def publish_tweet_with_pic text, pic
+      weibo_client.statuses.upload text, open(pic)
+    end
+
+    def valid_image?(url)
+      begin
+        url = URI.parse(url)
```

Comment:
What happens in weibo if we try to use a pic URL to an invalid pic?  Could we just check that it's a valid URL and then send it, and let them error if it doesn't work?